### DISCUSSION TITLE
Magic  button fail?

### DISCUSSION THREAD

**Discussion by N4N0-pa** [UzixLS/zx-sizif-512#60] [Q&A]
I have built 2 Rev. E1 boards and when I press the magic button the menu appears but I can't do anything because it disappears again after a couple of seconds on both boards, does anyone know why this could be? Also, if I insert a micro SD, the screen goes black when starting and if I press reset, strange characters appear on the screen.
Thanks in advance

**Comment by UzixLS** [comment]
Try to replace SRAM and Z80 chips. Current CPLD firmware version requires CMOS Z80 CPU (as in the BOM) and may fail with NMOS ones.

**Reply by N4N0-pa** [reply]
I have tried several cpu and several sram with the same result, the cpu that I have installed is exactly the one that appears in the BOM, can the 74hcu04 influence it? I ordered 10 from China and only 2 work, even though according to the xgpro they are all correct

**Reply by UzixLS** [reply]
Many chinese sellers sells remarked or even broken ICs. 74hcu04 may actually be 74hc04, which badly suitable in crystal generator circuit.

**Reply by N4N0-pa** [reply]
Thank you very much Eugene, I have managed to get the menu to work in both, in one it was a bad solder that generated a false key press and in the other the sram as you indicated, regarding the 74hcu04 I will have to order from a more trustworthy website

**Reply by AtomicRPM** [reply]
> Many chinese sellers sells remarked or even broken ICs. 74hcu04 may actually be 74hc04, which badly suitable in crystal generator circuit.

correct. The resulting frequency produced by the 28 MHz crystal at Y1 will be dramatically lower.  A crude work around when using Alliexpress fake 74HCU04s is to remove the 22pf capacitor C22.   This will correct the frequency.
